src/quickfix.c

```diff
@@ -2788,59 +2788,65 @@ qf_list(exarg_T *eap)
     {
 	if ((qfp->qf_valid || all) && idx1 <= i && i <= idx2)
 	{
-	    msg_putchar('\n');
-	    if (got_int)
-		break;
-
 	    fname = NULL;
-	    if (qfp->qf_module != NULL && *qfp->qf_module != NUL)
-		vim_snprintf((char *)IObuff, IOSIZE, "%2d %s", i, (char *)qfp->qf_module);
-	    else {
-		if (qfp->qf_fnum != 0
-				&& (buf = buflist_findnr(qfp->qf_fnum)) != NULL)
-		{
-		    fname = buf->b_fname;
-		    if (qfp->qf_type == 1)	/* :helpgrep */
-			fname = gettail(fname);
-		}
-		if (fname == NULL)
-		    sprintf((char *)IObuff, "%2d", i);
-		else
-		    vim_snprintf((char *)IObuff, IOSIZE, "%2d %s",
-								i, (char *)fname);
+	    if (qfp->qf_fnum != 0
+			      && (buf = buflist_findnr(qfp->qf_fnum)) != NULL)
+	    {
+		fname = buf->b_fname;
+		if (qfp->qf_type == 1)	/* :helpgrep */
+		    fname = gettail(fname);
 	    }
-	    msg_outtrans_attr(IObuff, i == qi->qf_lists[qi->qf_curlist].qf_index
+	    /* apply :filter /pat/ */
+	    if (cmdmod.filter_regmatch.regprog == NULL
+		    || (fname != NULL && !message_filtered(fname))
+		    || (qfp->qf_module != NULL && !message_filtered(qfp->qf_module)))
+	    {
+		msg_putchar('\n');
+		if (got_int)
+		    break;
+
+		if (qfp->qf_module != NULL && *qfp->qf_module != NUL)
+		    vim_snprintf((char *)IObuff, IOSIZE, "%2d %s", i, (char *)qfp->qf_module);
```

Comment:
Some of these lines are exceeding 80 characters, can you move the arguments to the next line?